### PR TITLE
Chore: Use `builder-support` and update docs.

### LINKED ISSUE #7: Extract `util/dev.js` / prod/dev dependencies Archetype publishing
From @chaseadamsio 

If you decide to go either route, it might make sense to abstract `util/dev.js` into its own package so that anyone who wanted to create builder archetypes could just use it without having to know how to use the script or update it whenever you decide to update it...it might even make sense to make it a part of `builder` as a task, something like: `builder publish archetype` inside of a repository that would publish your archetype and the `-dev` version for you.

### LINKED ISSUE #8: ARCHETYPE-dev: Copy README.md on publish.
Ignored by `.gitignore`, included with `.npmignore`

### PR DESCRIPTION
**Pull request by ryan-roemer**
Fixes #7
Fixes #8 
- Switches to https://github.com/FormidableLabs/builder-support for generating `dev/package.json`
- Does a full README + gitignore + dev/package.json generation
- Update README based on a version of `builder` that has `builder help <archetype>` fixed per https://github.com/FormidableLabs/builder/pull/53

This PR should supersede existing PR #20 (sorry for the wild goose chase Per!)

/cc @coopy @exogen @chaseadamsio

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -13,7 +13,7 @@
   },
   "homepage": "https://github.com/FormidableLabs/builder-react-component",
   "scripts": {
-    "npm:postinstall": "node -e \"require('fs').stat('lib', function(e,s){process.exit(e || !s.isDirectory() ? 1 : 0)})\" || builder run build-lib",
+    "npm:postinstall": "cd lib || builder run build-lib",
```
Comment (ryan-roemer) [inline package.json:16]: NOTE: Going for @exogen 's shorter, better version of this command.

Let me know if anyone feels we should split out to a separate PR...

**Comment by ryan-roemer** [comment]
@chaseadamsio -- Here it is! Automatic support for `-dev`

@boygirl -- We'll want to integrate this into the victory archetype too.

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -21,10 +21,11 @@ Run `npm run builder:check`
 
 ## Documentation
 
-To generate the usage snippet in the `README.md`, run:
+To generate the usage snippet in the `README.md`, first go to a project
+that _uses_ the archetype and then run:
```
Comment (exogen) [inline CONTRIBUTING.md:25]: Hoping this can be moved to `builder-support` eventually so this isn't necessary (the copy & paste part I don't mind; switching to a project & installing your updated archetype is a hassle).

**Comment by exogen** [comment]
LGTM, though looking forward to generating those docs in an easier way! :+1:

File: CONTRIBUTING.md (hunk 2 above)
Comment (ryan-roemer) [inline CONTRIBUTING.md:25]: @exogen -- Already ticketed as https://github.com/FormidableLabs/builder/issues/12

**Comment by ryan-roemer** [comment]
Published as `builder-react-component@0.1.3` and `builder-react-component-dev@0.1.3`
